### ISSUE TITLE
Row that is already selected does not open rec details on click

### ISSUE THREAD

**Issue by ericksod** [cloud-carbon-footprint/cloud-carbon-footprint#496]
**Context**
As a user looking to make recommended optimizations
I want to click to view the recommendation details
So that I can be fully informed

**Expected behavior**
- [x] When I click on a row in the recs table, the details panel pops out
- [x] When the recommendations detail panel is open and I click again on the row that is already selected, the panel closes
- [x] When a row is selected and the info panel is closed, and I click the selected row again, the info panel pops out
- [ ] When I have a row selected and click on another row, the info panel pops out reflecting the newly selected row
- [x] When the info panel is open and I click anywhere on the page (not the panel itself) the panel will close

**Current Behavior**
- When I close the info panel and click the selected row again, the info panel does not pop out

**Comment by jeff-zinger** [comment]
Hi there, I'd like to pick this up. I should have time next week to do it.

**Comment by jeff-zinger** [comment]
Hello, just confirming that I will be only able to complete 4 out of the 5 ACs. The last one - closing the panel when clicking elsewhere - doesn't seem practical to complete with the current implementation details of the side panel. We are currently using a "permanent" side panel in order to not have it interfere with the rest of the page e.g. allow scrolling.

I tried to see if switching it to a "temporary" side panel would allow me to complete the 5th AC and also maintain the current behaviour but was having difficulty making sure the panel would stay open if clicking on the actual panel itself - which is also part of the AC. This is due to the way the DOM is structured in the "temporary" implementation. Not only that, but this would only be possible if we kept the shadow overlay, which would mean you wouldn't be able to interact with the rest of the page while the panel is open - which is probably like that by design.
